### PR TITLE
WIP - Feat/replace yaml list

### LINKED ISSUE #21: Allow updating of yaml lists
- should we support adding an entry? And therefore also deleting?
- or just replacing the whole list with a new one?

### PR DESCRIPTION
**Pull request by joachimprinzbach**
This test shows the example for replacing the whole list - seems to work out of the box, only intending is missing

closes #21

### CODE REVIEW

File: tests/test_yaml_util.py [hunk 1]
```diff
@@ -65,3 +65,39 @@ def test_update_yaml_file(self):
             self.assertEqual(stream.readline(), "    y:\n")
             self.assertEqual(stream.readline(), "      z: foo_y_z\n")
             self.assertEqual(stream.readline(), "")
+
+        update_yaml_file(test_file, "newlist", ["val1", "val2"])
```
Comment (joachimprinzbach) [inline tests/test_yaml_util.py]: this one will lead to the new prop "newList"

File: tests/test_yaml_util.py [hunk 2]
```diff
@@ -65,3 +65,39 @@ def test_update_yaml_file(self):
...
+            self.assertEqual(stream.readline(), "    z: foo_z\n")
+            self.assertEqual(stream.readline(), "    y:\n")
+            self.assertEqual(stream.readline(), "      z: foo_y_z\n")
+            self.assertEqual(stream.readline(), "newlist:\n")
+            # TODO: These values should be intendet
+            self.assertEqual(stream.readline(), "- val1\n")
+            self.assertEqual(stream.readline(), "- val2\n")
+            self.assertEqual(stream.readline(), "")
+
+        update_yaml_file(test_file, "newlist", ["new_val1", "val2"])
```
Comment (joachimprinzbach) [inline tests/test_yaml_util.py]: Now the list will get replaced, the first entry is different

File: tests/test_yaml_util.py [hunk 3]
```diff
@@ -65,3 +65,39 @@ def test_update_yaml_file(self):
...
+            self.assertEqual(stream.readline(), "# comment\n")
+            self.assertEqual(stream.readline(), "  b:\n")
+            self.assertEqual(stream.readline(), "    d: 1 # comment\n")
+            self.assertEqual(stream.readline(), "    c: '2' # comment\n")
+            self.assertEqual(stream.readline(), "  x:\n")
+            self.assertEqual(stream.readline(), "    z: foo_z\n")
+            self.assertEqual(stream.readline(), "    y:\n")
+            self.assertEqual(stream.readline(), "      z: foo_y_z\n")
+            self.assertEqual(stream.readline(), "newlist:\n")
+            # TODO: These values should be intendet
```
Comment (joachimprinzbach) [inline tests/test_yaml_util.py]: list entries should be intended

**Comment by joachimprinzbach** [comment]
I think we can delete this one

**Comment by christiansiegel** [comment]
ack

**Pull request closed without merging**